File: voyageone-service2/service2-model/src/main/java/com/voyageone/service/bean/cms/ComplexMappingBean.java
package com.voyageone.service.bean.cms;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Leo on 15-12-9.
 */
public class ComplexMappingBean extends com.voyageone.service.bean.cms.MappingBean {
    private List<com.voyageone.service.bean.cms.MappingBean> subMappings;
    private String masterPropId;

    public ComplexMappingBean() {
        mappingType = MAPPING_COMPLEX;
        subMappings = new ArrayList<>();
    }

    public List<com.voyageone.service.bean.cms.MappingBean> getSubMappings() {
        return subMappings;
    }

    public void setSubMappings(List<com.voyageone.service.bean.cms.MappingBean> subMappings) {
        this.subMappings = subMappings;
    }

    public String getMasterPropId() {
        return masterPropId;
    }

    public void setMasterPropId(String masterPropId) {
        this.masterPropId = masterPropId;
    }

    public void addSubMapping(com.voyageone.service.bean.cms.MappingBean subMapping) {
        subMappings.add(subMapping);
    }
}

File: voyageone-static/2.0/develop/modules/cms/models/ComplexMappingBean.js
define([
    'modules/cms/enums/MappingTypes'
], function (MappingTypes) {

    /**
     * 构造一个和 Java 端 com.voyageone.cms.service.bean.ComplexMappingBean 一样的实例
     * @constructor
     * @extends {MappingBean}
     * @param {string} platformPropId
     * @param {string} masterPropId
     * @param {MappingBean[]} subMapping
     */
    function ComplexMappingBean(platformPropId, masterPropId, subMapping) {
        /**
         * 平台属性 ID
         * @type {string}
         */
        this.platformPropId = platformPropId;
        /**
         * 当前匹配类型, 对应 MappingTypes 的 SIMPLE
         * @type {string}
         */
        this.mappingType = MappingTypes.COMPLEX_MAPPING;
        /**
         * 主数据属性 ID
         * @type {string}
         */
        this.masterPropId = masterPropId;
        /**
         * 继承 MappingBean 的 Mapping 关系集合
         * @type {MappingBean[]}
         */
        this.subMapping = subMapping;
    }

    return ComplexMappingBean;

});